### PR TITLE
azurerm_kubernetes_cluster  - private_cluster_public_fqdn_enabled is no longer force new

### LINKED ISSUE #13099: Kubernetes cluster must be replaced because of private_cluster_public_fqdn_enabled
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

Terraform - v1.0.5
azurerm - v2.73.0

### Affected Resource(s)

* `azurerm_kubernetes_cluster`

### Terraform Configuration Files

```hcl
terraform {
  required_providers {
    azurerm = {
      source = "hashicorp/azurerm"
      version = "2.73.0"
    }
  }
}

provider "azurerm" {
  features {}
}

locals {
  aks_tags_dev = {
    Owner             = "__owner__"
    CreatedBy         = "__createdby__"
    EnvironmentType   = "dev"
    CostCenter        = "4600-9000"
    Website           = "__website__"
  }
}

resource "azurerm_kubernetes_cluster" "aks-web-dev-001" {
  name                                  = "aks-web-dev-001"
  location                              = "__location__"
  resource_group_name                   = "rg-web-dev-001"
  dns_prefix                            = "aks-web-dev-001-dns"
  kubernetes_version                    = "1.19.11"
  node_resource_group                   = "rg-web-dev-nodes-001"

  tags = local.aks_tags_dev

  identity {
    type             = "SystemAssigned"
  }

  default_node_pool {
    name                          = "sysb4ms001"
    vm_size                       = "Standard_B4ms"
    node_count                    = 1
    max_pods                      = 60
    only_critical_addons_enabled  = true
    orchestrator_version          = "1.19.11"
    vnet_subnet_id                = "__aksSubnetId__"

    tags = local.aks_tags_dev
  }

  linux_profile {
    admin_username = "azureuser"

    ssh_key {
      key_data = "__aksSshKey__"
    }
  }

  network_profile {
    dns_service_ip        = "10.201.0.10"
    docker_bridge_cidr    = "172.201.0.1/16"
    network_plugin        = "azure"
    network_policy        = "calico"
    service_cidr          = "10.201.0.0/16"
  }

  role_based_access_control {
    enabled = true
  }
}
```

### Debug Output

Debug contains secrets, so would prefer a private method if necessary for sharing.

### Expected Behaviour

The Costcenter tag gets replaced by the CostCenter tag without causing downtime on the cluster.

Output should be something like:
```
Terraform will perform the following actions:

  # azurerm_kubernetes_cluster.aks-web-dev-001 will be updated in-place
  ~ resource "azurerm_kubernetes_cluster" "aks-web-dev-001" {
    ...
      ~ tags                       = {
          + "CostCenter"      = "4600-9000"
          - "Costcenter"      = "4600-9000" -> null
            # (3 unchanged elements hidden)
        }
    }
```

### Actual Behaviour

The whole cluster gets replaced.

Actual output is something like:
```
Terraform will perform the following actions:

  # azurerm_kubernetes_cluster.aks-web-dev-001 must be replaced
-/+ resource "azurerm_kubernetes_cluster" "aks-web-dev-001" {
      + private_cluster_public_fqdn_enabled = false # forces replacement
    }
```

The same configuration works as expected on azurerm 2.72.0

### Steps to Reproduce

1. `terraform plan`

### PR DESCRIPTION
**Pull request by hieumoscow**
Fix #13099, to do in place update for `private_cluster_public_fqdn_enabled`

### CODE REVIEW

**Comment by LaurentLesle** [comment]
would be good if someone can fast-track the review and get it in the next release as it is impacting customers with private clusters who need to upgrade to azurerm ~>= 2.73

File: internal/services/containers/kubernetes_cluster_resource.go [hunk 1]
```diff
@@ -593,7 +593,7 @@ func resourceKubernetesCluster() *pluginsdk.Resource {
 				Type:     pluginsdk.TypeBool,
 				Optional: true,
 				Default:  false,
-				ForceNew: true,
+				ForceNew: false,
```
Comment (aristosvo) [inline internal/services/containers/kubernetes_cluster_resource.go:596]: It is better for remove the `ForceNew` annotation here at all
```suggestion
```

**Review by aristosvo — COMMENTED** [review]
It looks good to me besides the comment, but I'm not a maintainer tbh. I'll run the specific AccTest now to check if it is indeed working, will let you know.

**Review by katbyte — CHANGES_REQUESTED** [review]
Thanks for the change @hieumoscow - could we update the docs to reflect this change and add/update the update test to ensure this continues to work? thanks!

File: internal/services/containers/kubernetes_cluster_resource.go (hunk 1 above)
Comment (hieumoscow) [reply internal/services/containers/kubernetes_cluster_resource.go]: Thanks @aristosvo, I have added the change per your comments in https://github.com/hashicorp/terraform-provider-azurerm/pull/13413/commits/8c29589dee193c3c36ee7708d0fc56ff433e17dc

**Comment by hieumoscow** [comment]
@katbyte I have updated per @aristosvo suggestion, I do not think we need to update the doc as there was no _"Changing this forces a new resource to be created."_ phrase there before.

https://github.com/hieumoscow/terraform-provider-azurerm/blame/main/website/docs/r/kubernetes_cluster.html.markdown#L129

**Comment by hieumoscow** [comment]
Ran the test locally, all passed!

**Review by katbyte — APPROVED** [review]
Thanks @hieumoscow! LGTM now 🏗️

**Pull request merged**